server/php-master.cpp

```diff
@@ -1520,28 +1513,29 @@ WorkerType run_master() {
 
     me->generation = generation;
 
-    if (to_kill != 0 || to_run != 0 || job_workers_to_kill != 0 || job_workers_to_run != 0) {
-      vkprintf(1, "[to_kill = %d] [to_run = %d] [job_workers_to_kill = %d] [job_workers_to_run = %d]\n", to_kill, to_run, job_workers_to_kill, job_workers_to_run);
+    if (general_worker_to_kill != 0 || general_worker_to_run != 0 || job_workers_to_kill != 0 || job_workers_to_run != 0) {
+      tvkprintf(master_process, 2, "[general_worker_to_kill = %d] [general_worker_to_run = %d] [job_workers_to_kill = %d] [job_workers_to_run = %d]\n",
+                general_worker_to_kill, general_worker_to_run, job_workers_to_kill, job_workers_to_run);
     }
 
     for (int i = 0; i < job_workers_to_kill; ++i) {
       kill_worker(WorkerType::job_worker);
     }
     for (int i = 0; i < job_workers_to_run; ++i) {
       if (run_worker(WorkerType::job_worker)) {
-        tvkprintf(job_workers, 1, "launched new job worker with pid = %d\n", pid);
         return WorkerType::job_worker;
       }
     }
 
-    while (to_kill-- > 0) {
+    for (int i = 0; i < general_worker_to_kill; ++i) {
       kill_worker(WorkerType::general_worker);
     }
-    while (to_run-- > 0 && !failed) {
+    for (int i = 0; i < general_worker_to_run; ++i) {
```

Comment:
You forgot about `!failed` condition

Reply:
Return and add it to `job_workers_to_run`. Is it correct?

Reply:
Let's keep it exactly like before and remove from `job_workers_to_run`

Reply:
ok